modules/Bio/EnsEMBL/IO/Translator/SampleGenotypeFeature.pm

```diff
@@ -0,0 +1,268 @@
+
+my %field_callbacks = ('chrom'  => 'chrom',
+                       'pos'    => 'pos',
+                       'id'     => 'id',
+                       'ref'    => 'ref',
+                       'alt'    => 'alt',
+                       'qual'   => 'qual',
+                       'filter' => 'filter',
+                       'info'   => 'info',
+                       'format' => 'format'
+                      );
+                      
+my %info_methods = ('AA' => 'ancestral_allele',
+                    #'AC'  => 'allele_count',
+                    #'AF'  => 'allele_frequency',
+                    #'AN'  => 'total_allele_number',
+                    'MA'  => 'minor_allele',
+                    'MAF' => 'minor_allele_frequency',
+                    'MAC' => 'minor_allele_count',
+                    'NS'  => 'samples_with_data'
+                   );
+
+=head2 new
+
+    Returntype   : Bio::EnsEMBL::IO::Translator::SampleGenotypeFeature
+
+=cut
+
+sub new {
+    my ($class, $info_list, $formats_list, $samples_list) = @_;
+  
+    my $self = $class->SUPER::new();
+    
+    # Once we have the instance, add our customized callbacks
+    # to the translator
+    $self->add_callbacks(\%field_callbacks);
+
+    if ($samples_list && scalar(@$samples_list) > 0) {
+      $self->{'samples_list'} = $samples_list;
+    }
+    
+    $self->{'info_list'}    = $info_list;
+    $self->{'formats_list'} = $formats_list;
+    
+
+    return $self;
+}
+
+sub set_samples_list {
+  my $self = shift;
+  my $samples_list = shift;
+  
+  if ($samples_list && scalar(@$samples_list) > 0) {
+    $self->{'samples_list'} = $samples_list;
+  }
+}
+
+sub get_samples_list {
+  my $self = shift;
+  
+  return ($self->{'samples_list'} && scalar(@{$self->{'samples_list'}}) > 0) ? $self->{'samples_list'} : [];
+}
+
+sub samples_genotypes {
+  my $self = shift;
+  my $object = shift;
+  
+  my $samples_list = $self->get_samples_list();
+  
+  die "Error! You need to define a list of samples as argument.\n" if (scalar(@$samples_list) < 1);
+  
+  # Transform the alleles in 0 | 1 | 2..., e.g. 0 if it matches the reference, 1 if it matches the alt 1, ...
+  my @alleles = split(/\//,$self->alleles($object));
+  my %gen_al;
+  for (my $i = 0;$i < @alleles;$i++) {
+    $gen_al{$alleles[$i]} = $i;
+  }
+
+  my %s_gen = map { $_->sample->name => { 'genotype' => $self->genotype_vcf(\%gen_al,$_->genotype), 
+                                          'phased'   => ($_->phased) ? $_->phased : 0
+                                        }
+                  } @{$object->variation->get_all_SampleGenotypes()};
+  
+  if ($samples_list and scalar(@$samples_list) > 0) {
+    my %new_s_gen = map { $_ => $s_gen{$_}} @$samples_list;
+    %s_gen = %new_s_gen;
+  }
+  
+  return \%s_gen;
+}
+
+sub chrom {
+  my $self = shift;
+  my $object = shift;
+  
+  return $self->seqname($object);
+}
+
+sub pos {
+  my $self = shift;
+  my $object = shift;
+  
+  return $self->start($object);
```

Comment:
What do you mean exactly ? It is calling the methods from "Bio::EnsEMBL::IO::Translator::VariationFeature"